### DISCUSSION TITLE
How to compile this source

### DISCUSSION THREAD

**Discussion by fahzafahmi** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2690] [Q&A]
Hello, i would like to ask a question of how to compile this LCD firmware?

Because i thought it would be similar with compiling Marlin firmware, but i found it's not similar at all

Looking forward for a tutorial of it.

**Comment by radek8** [comment]
Yes it is similar to Marlin.
Use Visual Studio Code + PlatformIO IDE.
If you can't do that, use a precompiled bin file.
You complete the configuration with the ini file.

**Comment by rzboril** [comment]
I buld with Visual Studio and failed with platform iO : 
Compiling .pio\build\BIGTREE_GD_TFT50_V3_0\src\src\User\Hal\buzzer.o
Compiling .pio\build\BIGTREE_GD_TFT50_V3_0\src\src\User\Hal\gd32f20x\GPIO_Init.o
TFT\src\User\Hal\STM32_USB_OTG_Driver\src\usb_core.c: In function 'USB_OTG_WritePacket':
TFT\src\User\Hal\STM32_USB_OTG_Driver\src\usb_core.c:178:7: warning: 'packed' attribute ignored for type 'uint32_t *' {aka 'long unsigned int *'} [-Wattributes]
  178 |       USB_OTG_WRITE_REG32( fifo, *((__packed uint32_t *)src) );
      |       ^~~~~~~~~~~~~~~~~~~
TFT\src\User\Hal\STM32_USB_OTG_Driver\src\usb_core.c: In function 'USB_OTG_ReadPacket':
TFT\src\User\Hal\STM32_USB_OTG_Driver\src\usb_core.c:204:5: warning: 'packed' attribute ignored for type 'uint32_t *' {aka 'long unsigned int *'} [-Wattributes]
  204 |     *(__packed uint32_t *)dest = USB_OTG_READ_REG32(fifo);
      |     ^
Compiling .pio\build\BIGTREE_GD_TFT50_V3_0\src\src\User\Hal\gd32f20x\HAL_Flash.o
..
